test/state/state.cpp

```diff
@@ -117,19 +117,22 @@ evmc_message build_message(const Transaction& tx, int64_t execution_gas_limit) n
 }
 }  // namespace
 
+void finalize(State& state, evmc_revision rev)
+{
+    std::erase_if(state.get_accounts(), [rev](const std::pair<const address, Account>& p) noexcept {
```

Comment:
This is incorrect. It seems the implementation practice is to split handling selfdestructed accounts from empty-but-touched. We should delete selfdestructed after each transaction and clear empty-but-touched in the very end after applying block reward and withdrawals. See https://github.com/ethereum/yellowpaper/issues/885.